sftp/sftp.go

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2022-2023 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package sftp
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+)
+
+type (
+	// LogType defined various types of logs and errors
+	// that can happen within the SFTP implementation
+	LogType string
+)
+
+const (
+	missingConnectionHandlerFunction = "New connection handler is not defined"
```

Comment:
Make these exported `var ErrMissingConnectionHandlerFunction = errors.New("new connection handler is not defined")` so callers can compare.